pkg/jwt/basic/provider.go

```diff
@@ -0,0 +1,35 @@
+package basic
+
+import (
+	"github.com/hellofresh/janus/pkg/config"
+	"github.com/hellofresh/janus/pkg/jwt/provider"
+)
+
+func init() {
+	provider.Register("basic", Provider{})
+}
+
+// Provider abstracts the authentication for github
+type Provider struct {
+	provider.Verifier
+}
+
+// Build acts like the constructor for a provider
+func (gp Provider) Build(config config.Credentials) provider.Provider {
```

Comment:
I'd rather `New` over `Build` but if this convention is being used that's fine

Reply:
The provider is acting as an [Abstract Factory](https://sourcemaking.com/design_patterns/abstract_factory) where it will create an object based on the given configurations. Do you think it should be `New` anyway?

Reply:
Hm now I see you are right, better keep any `new` inside of the factory, thank you!